src/components/Button/Button.tsx

```diff
@@ -1,13 +1,80 @@
-import { Text, TouchableOpacity } from 'react-native'
+import React from 'react'
+import { MaterialIcons } from '@expo/vector-icons'
+import { View } from 'react-native'
+import {
+  Button as PaperButton,
+  ButtonProps as PaperButtonProps,
+} from 'react-native-paper'
 
-import { ButtonProps } from './Button.types'
+export interface ButtonProps
+  extends Omit<PaperButtonProps, 'icon' | 'children'> {
+  title?: string | React.ReactNode
+  onPress?: () => void
+  materialIconName?: keyof typeof MaterialIcons.glyphMap
+  icon?: React.ReactNode
+  alignIcon?: 'left' | 'right'
+  loading?: boolean
+}
+
+export default function Button({
+  title,
+  icon,
+  materialIconName,
+  alignIcon = 'left',
+  loading = false,
+  onPress,
+  mode = 'contained',
+  ...rest
+}: ButtonProps) {
+  const handleIcon = React.useCallback(() => {
+    if (icon) {
+      return icon
+    }
+    if (materialIconName) {
+      return <MaterialIcons name={materialIconName} size={20} color={'#fff'} />
+    }
+    return undefined
+  }, [icon, materialIconName])
```

Comment:
Pq uma função? Não podemos usar um ternário, ou algo como [ts-pattern](https://github.com/gvergnaud/ts-pattern)?

```suggestion
export default function Button({
  title,
  icon: IconProp,
  materialIconName,
  alignIcon = 'left',
  loading = false,
  onPress,
  mode = 'contained',
  ...rest
}: ButtonProps) {
  const Icon = IconProp !== undefined ? <Icon /> : <MaterialIcons name={materialIconName} size={20} color="#fff" />
```

Reply:
+1 pra introdução do ts-pattern

Reply:
Eu curto ternário mas acho que função é algo mais legível, mas então nesse caso usarei o ts-pattern. Valeu!